bodhi/tests/server/services/test_updates.py

```diff
@@ -932,6 +932,83 @@ def test_provenpackager_request_update_with_none_ci(self, publish, *args):
             res.json_body['errors'][0]['description'],
             config.get('not_yet_tested_msg'))
 
+    @mock.patch(**mock_taskotron_results)
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_provenpackager_request_update_with_gw_satisfied(self, publish, *args):
+        """Ensure provenpackagers can request changes for any update which
+        satisfies Greenwave policies"""
+        nvr = u'bodhi-2.1-1.fc17'
+        user = User(name=u'bob')
+        self.db.add(user)
+        group = self.db.query(Group).filter_by(name=u'provenpackager').one()
+        user.groups.append(group)
+        settings = self.app_settings.copy()
+        settings['greenwave_enabled'] = True
+        self.db.commit()
+
+        app = TestApp(main({}, testing=u'bob', session=self.db, **settings))
```

Comment:
It seems that you might be creating an app to get a setting to be a particular way here. The ```TestApp``` is known to have a memory leak in it and is the source of much of the memory usage when running Bodhi's test suite. Instead, it would be better to use ```mock.patch.dict``` on ```bodhi.server.config.config``` to get settings to be a particular way.

There should be a ```self.app``` that you can use that is a singleton for the test suite to help keep memory usage down. There are still a few places in the code that create apps like this and we hope to eliminate the rest of them in the future.